### ISSUE TITLE
Context window too long and chats not clearing

### ISSUE THREAD

**Issue by ragsntatters** [stackblitz/bolt.new#2283]
### Describe the bug

Soo the app is keeping all the previous chat logs and even if i ask it to do a simple task it says that I'm using 200k tokens+ Seems like a ripoff. Things started so well, what is going on? I'm like 100M+ tokens deep and the app is unusable

### Link to the Bolt URL that caused the error

bolt.new

### Steps to reproduce

Check your app

### Expected behavior

For it to work

### Platform

- OS: [e.g. macOS, Windows, Linux]
- Browser: [e.g. Chrome, Safari, Firefox]
- Version: [e.g. 91.1]

**Comment by jerekel87** [comment]
Same issue with me, are you getting this error 

prompt is too long: 207587 tokens > 200000 maximum ?? or something similar?

**Comment by ragsntatters** [comment]
> Same issue with me, are you getting this error
> 
> prompt is too long: 207587 tokens > 200000 maximum ?? or something similar?

hey yup that's the one, at first i thought i was doing something wrong but naa, it's been shaky for the past few days and unusable today

**Comment by jerekel87** [comment]
I just left for like 3 hours came back, it allowed me to do 2 prompts without any issues and the 3rd prompt I got hit with the error again lmfao

I think this Bolt app knows when someone is doing a lot to build some type of app and this is there way of slowing people down. There is no other explanation behind why it does this. I've spent around $400 on tokens right now and I still have so much to develop and yet I'm getting hit with these stupid errors just like everyone else. This should be a priority on their list of things to fix, but it doesn't seem to be for some reason.

**Comment by ragsntatters** [comment]
yeah I think either a) they're trying to make it faster / more accurate by
getting it to cache all the data it sees which then clogs it up after a
couple of edits or b) the bug is a feature to get n00bs to get more tokens
and the reason why it isn't being fixed is a combo of the two. I've also
noticed that it keeps doing the same fix attempts / will redo the same work
repeatedly which initially I thought was required but now I think it's dodgy

Raghav Saxena
Marketing Director

[image: Logo] <https://www.digigo.co.nz/>

Digigo Marketing
Auckland, NZ
phone: 0800TOGROW
mobile: +642102489262
***@***.***
digigo.co.nz
[image: Facebook icon] <https://www.facebook.com/DigigoNZ>

**Comment by jerekel87** [comment]
Yeah a lot of weird things are happening here and if your the owner / developer of an Ai app this mind blowing... why would you not dump all your resources in to fixing this one major problem? Btw this isn't even their biggest problem, I can't do any edits / changes as I'm being hit with a prompt is too long: 200067 tokens > 200000 maximum for just trying to deploy or asking it to change a button color lmfao

Even when I refresh, it some times works and when it does work it works for one command / request after that it shows the same error message. 

I can't get enough of this app, I've spent $400 on tokens building what I'm building and for some reason Bolt.new wont let me finish building it because it's saying prompt is too long: 200067 tokens > 200000 maximum (prompt is not too long when you ask the fucking Ai to change the color of a button lol)
